GitUI/CommandsDialogs/ToolStripPushButton.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Windows.Forms;
+using GitCommands;
+using GitCommands.Git;
+using GitUI.Properties;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs
+{
+    public class ToolStripPushButton : ToolStripButton
+    {
+        private readonly TranslationString _push = new TranslationString("Push");
+
+        private readonly TranslationString _aheadCommitsToPush =
+            new TranslationString("{0} new commit(s) will be pushed");
+
+        private readonly TranslationString _behindCommitsTointegrateOrForcePush =
+            new TranslationString("{0} commit(s) should be integrated (or will be lost if force pushed)");
+
+        private IAheadBehindDataProvider _aheadBehindDataProvider;
+        private bool _supportsAheadBehindData;
+
+        public void Initialize(IAheadBehindDataProvider aheadBehindDataProvider, bool supportsAheadBehindData)
+        {
+            _aheadBehindDataProvider = aheadBehindDataProvider;
+            _supportsAheadBehindData = supportsAheadBehindData;
+            ResetToDefaultState();
+        }
+
+        public void DisplayAheadBehindInformation(string branchName)
+        {
+            if (!_supportsAheadBehindData)
+            {
+                return;
+            }
+
+            ResetToDefaultState();
+
+            var aheadBehindData = _aheadBehindDataProvider.GetData(branchName);
+            if (aheadBehindData == null || aheadBehindData.Count < 1 || !aheadBehindData.ContainsKey(branchName))
+            {
+                return;
+            }
+
+            var data = aheadBehindData[branchName];
+
+            if (AppSettings.ShowAheadBehindData)
```

Comment:
The ahead/behind info is calculated with a separate Git command also if false.
Discussion: If calculate for all local branches, there can be many Git commands. Should this be a separate setting?